test/unit/org/apache/cassandra/io/sstable/SSTableReaderTest.java

```diff
@@ -1242,6 +1242,7 @@ private <R extends SSTableReader & IndexSummarySupport<R>> void testIndexSummary
         }
         R reopen = (R) SSTableReader.open(store, sstable.descriptor);
         assert reopen.getIndexSummary().getSamplingLevel() == sstable.getIndexSummary().getSamplingLevel() + 1;
+        reopen.selfRef().release();
```

Comment:
Is this to deal with the leaks? It needs to be in try/finally around everything since the ref is acquired.